### PR TITLE
Add vss commitment to participant demo and generate key package

### LINKED ISSUE #32: Add vss_commitments to Participant demo
## Context

The Participant needs the commitments to build a Secret Share and generate a Key Package

## Todo

- [x] Accept vss_commitment as input
- [x] Validate input
- [x] Generate key package

### PR DESCRIPTION
**Pull request by natalieesk**
This closes #32

### CODE REVIEW

File: participant/src/lib.rs [hunk 1]
```diff
@@ -51,13 +60,94 @@ pub fn request_inputs(input: &mut impl BufRead, logger: &mut dyn Logger) -> Resu
...
     input.read_line(&mut signing_share_input).unwrap();
 
-    let signing_share =
-        <[u8; 32]>::from_hex(signing_share_input.trim()).map_err(|_| Error::MalformedSigningKey)?;
+    let signing_share = SigningShare::from_bytes(
+        <[u8; 32]>::from_hex(signing_share_input.trim()).map_err(|_| Error::MalformedSigningKey)?,
+    )
+    .unwrap(); // TODO: Need the correct error from frost
+
+    // TODO: Is extra validation needed here for public_key and signing_share or will that be resolved when used in generating key_packages etc.? Need to check
```
Comment (conradoplg) [inline participant/src/lib.rs:68]: Yep generating KeyPackage should validate the signature share.

File: participant/src/lib.rs [hunk 2]
```diff
@@ -51,13 +60,94 @@ pub fn request_inputs(input: &mut impl BufRead, logger: &mut dyn Logger) -> Resu
...
+
+    logger.log("Your verifiable secret sharing commitment:".to_string());
```
Comment (conradoplg) [inline participant/src/lib.rs:70]: The trusted dealer demo prints `"Your commitment"` so users might not recognize that's the thing they need to input here. My suggestion is making the two equal (maybe changing the trusted dealer to match this?)

**Review by conradoplg — CHANGES_REQUESTED** [review]
Looks good, thanks! There's one suggestion to make the prompt clearer and some comments that don't require any changes.

File: participant/src/lib.rs (hunk 2 above)
Comment (natalieesk) [reply participant/src/lib.rs:70]: Yes, good point.

File: participant/src/lib.rs (hunk 1 above)
Comment (natalieesk) [reply participant/src/lib.rs]: Removed comment [9c7d886](https://github.com/ZcashFoundation/frost-zcash-demo/pull/35/commits/9c7d886956b1599f46711d1b05d69059f0aacb1c)

File: participant/src/lib.rs (hunk 2 above)
Comment (natalieesk) [reply participant/src/lib.rs:73]: Done [9c7d886](https://github.com/ZcashFoundation/frost-zcash-demo/pull/35/commits/9c7d886956b1599f46711d1b05d69059f0aacb1c)

File: participant/src/lib.rs [hunk 3]
```diff
@@ -33,11 +41,12 @@ pub fn request_inputs(input: &mut impl BufRead, logger: &mut dyn Logger) -> Resu
 
     input.read_line(&mut public_key_input).unwrap();
 
-    let public_key =
-        <[u8; 32]>::from_hex(public_key_input.trim()).map_err(|_| Error::MalformedVerifyingKey)?;
+    let public_key = VerifyingShare::from_bytes(
+        <[u8; 32]>::from_hex(public_key_input.trim()).map_err(|_| Error::MalformedVerifyingKey)?,
+    )
+    .unwrap(); // TODO: Frost library needs to be updated with correct Error type
```
[the remark this answers is not in the document]
Comment (conradoplg) [reply participant/src/lib.rs]: Ahh I see what you mean, good point.

[the remark this answers is not in the document]
Comment (conradoplg) [reply participant/src/lib.rs]: Feel free to create a ticket to track the suggestion!

**Review by conradoplg — APPROVED** [review]

**Pull request merged**